Sources/ShopifyAcceleratedCheckouts/Wallets/ApplePay/ApplePayButton.swift

```diff
@@ -0,0 +1,233 @@
+    ///   - configuration: The configuration for Apple Pay
+    ///   - checkoutSuccessHandler: The handler to call on successful checkout
+    ///   - checkoutErrorHandler: The handler to call on checkout error
+    ///   - checkoutCancelHandler: The handler to call on checkout cancel
+    ///   - shouldRecoverFromErrorHandler: The handler to determine error recovery
+    ///   - clickLinkHandler: The handler to call when links are clicked
+    ///   - webPixelEventHandler: The handler to call for web pixel events
+    init(
+        identifier: CheckoutIdentifier,
+        label: PayWithApplePayButtonLabel,
+        configuration: ApplePayConfigurationWrapper,
+        eventHandlers: EventHandlers = EventHandlers()
+    ) {
+        controller = ApplePayViewController(
+            identifier: identifier,
+            configuration: configuration
+        )
+        controller.onComplete = eventHandlers.checkoutDidComplete
+        controller.onFail = eventHandlers.checkoutDidFail
+        controller.onCancel = eventHandlers.checkoutDidCancel
+        controller.onShouldRecoverFromError = eventHandlers.shouldRecoverFromError
+        controller.onClickLink = eventHandlers.checkoutDidClickLink
+        controller.onWebPixelEvent = eventHandlers.checkoutDidEmitWebPixelEvent
+        self.label = label
+    }
+
+    var body: some View {
+        PayWithApplePayButton(
+            label,
+            action: {
+                Task { await controller.startPayment() }
+            },
+            fallback: {
+                // content == nil ? Text("errors.applepay.unsupported") : content
+                Text("errors.applepay.unsupported".localizedString)
+            }
+        )
+        .walletButtonStyle()
+    }
+}
+
+// MARK: - Mock for Previews
+
+@available(iOS 17.0, *)
+let mockCommonConfiguration = ShopifyAcceleratedCheckouts.Configuration(
+    storefrontDomain: "my-shop.myshopify.com",
+    storefrontAccessToken: "asdb"
+)
+
+@available(iOS 17.0, *)
+let mockApplePayConfiguration = ShopifyAcceleratedCheckouts.ApplePayConfiguration(
+    merchantIdentifier: "merchantid",
+    supportedNetworks: [.amex, .discover, .masterCard, .visa],
+    contactFields: [.email, .phone]
+)
+
+@available(iOS 17.0, *)
+let mockShopSettings = ShopSettings(
+    name: "My Shop",
+    primaryDomain: Domain(
+        host: mockCommonConfiguration.storefrontDomain,
+        url: "https://\(mockCommonConfiguration.storefrontDomain)"
+    ),
+    paymentSettings: PaymentSettings(
+        countryCode: "US"
+    )
+)
+
+@available(iOS 17.0, *)
+let mockConfiguration = ApplePayConfigurationWrapper(
+    common: mockCommonConfiguration,
+    applePay: mockApplePayConfiguration,
+    shopSettings: mockShopSettings
+)
+
+@available(iOS 17.0, *)
+let mockLocale = Locale(identifier: "en")
+
+@available(iOS 17.0, *)
+let mockController = MockApplePayViewController(
+    identifier: .cart(cartID: "gid://Shopify/Cart/12345"),
+    configuration: mockConfiguration
+)
```

Comment:
I don't think these mocks belong here in the source code. Can we use define them in the sample app instead?

Reply:
Yeah preview in xcode is so unreliable. Let's remove 'em and consider another strategy for reintegrating them later 👍